src/core/image.js

```diff
@@ -525,7 +523,9 @@ var PDFImage = (function PDFImageClosure() {
           }
           if (this.needsDecode) {
             // Invert the buffer (which must be grayscale if we reached here).
-            assert(kind === ImageKind.GRAYSCALE_1BPP);
```

Comment:
looks like a good candidate for assert, if needsDecode == true is driven only by internal logic (and not use data)